File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Variable.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.streampark.console.core.entity;

import org.apache.streampark.common.constants.Constants;
import org.apache.streampark.console.base.mybatis.entity.BaseEntity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@Getter
@Setter
@EqualsAndHashCode(callSuper = true)
@TableName("t_variable")
public class Variable extends BaseEntity {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    @NotBlank(message = "{required}")
    private String variableCode;

    @NotBlank(message = "{required}")
    private String variableValue;

    @Size(max = 100, message = "{noMoreThan}")
    private String description;

    /** user id of creator */
    private Long creatorId;

    /** user name of creator */
    private transient String creatorName;

    @NotNull(message = "{required}")
    private Long teamId;

    private Boolean desensitization;

    public void dataMasking() {
        if (desensitization) {
            this.setVariableValue(Constants.DEFAULT_DATAMASK_STRING);
        }
    }
}

File: streampark-console/streampark-console-webapp-v2/src/service/api/resource/variable.ts
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import type {
  VariableDeleteParam,
  VariableListRecord,
  VariableParam,
} from '@/types/api/resource/variable/model/variableModel'
import type { BasicTableParams } from '@/types/api/model/baseModel'
import { request } from '../../http'

export function fetchVariableList(data: BasicTableParams & { variableCode?: string }) {
  return request.Post<VariableListRecord[]>('/variable/page', data)
}

export function fetchAddVariable(data: VariableParam) {
  return request.Post('/variable/post', data)
}

export function fetchUpdateVariable(data: VariableParam) {
  return request.Put('/variable/update', data)
}

export function fetchVariableDelete(data: VariableDeleteParam) {
  return request.Delete<{ status: string, message?: string }>('/variable/delete', data)
}

export function fetchCheckVariableCode(data: { variableCode: string }) {
  return request.Post<{ status: string, data?: boolean, message?: string }>('/variable/check/code', data)
}

export function fetchVariableInfo(data: { id: string }) {
  return request.Post<VariableListRecord>('/variable/show_original', data)
}

export function fetchDependApps(data: Recordable) {
  return request.Post<Recordable[] | { records?: Recordable[], total?: number }>('/variable/depend_apps', data)
}

export function fetchVariableAll(data?: { keyword?: string }) {
  return request.Post<VariableListRecord[]>('/variable/list', data ?? {})
}
